### ISSUE TITLE
Convert STRING to RID(LINK) in SQL not working?

### ISSUE THREAD

**Issue by Nacnudge** [orientechnologies/orientdb#6435]
## Expected behavior and actual behavior
**GOAL**: Convert string represention of an RID into a true RID to address referenced attributes/properties. 

Running a SQL method like `.convert("LINK")` off a property that has a string value of an `RID` should result in a `RID` "type"... but it remains as a `STRING` type - and as such is not traversable 

(eg. `XXX.convert("LINK").name` won't work)

## Steps to reproduce the problem
Run `XXX.convert('LINK')` method (where XXX is a STRING representation of an RID)

eg: (Class = ci, schemaless, cluster ID = 13)
2 records: 
```
#13:0 : {name:"First",rid2:"#13:1"}
#13:1 : {name:"Second",rid2:"#13:0"}
```

`select rid2.convert("LINK").name as linkedname from #13:0;  `

Should return `{ "linkedname":"Second" }`, _but it doesn't_

## Important Questions
- Is there another way to convert a string representation of an RID into a true RID?
- Shouldn't the above (`.convert("link")`) achieve this? 

### Running Mode
- [X] Remote

### Misc
- [X] Running a local test instance using console.sh

### OrientDB Version
- [X] v2.2.x - Please specify last number: 4

### Operating System
- [X] Linux

### Java Version
- [X] 8

**Comment by luigidellaquila** [comment]
Hi @Nacnudge 

I just pushed a fix on 2.2.x and develop branches. The fix will be released with 2.2.11

Thanks

Luigi

**Comment by Nacnudge** [comment]
Thanks @luigidellaquila!!
